zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/message/ProcessMessageSubscriptionCorrelateProcessor.java

```diff
@@ -109,6 +117,19 @@ final var record = command.getValue();
         stateWriter.appendFollowUpEvent(
             subscription.getKey(), ProcessMessageSubscriptionIntent.CORRELATED, record);
 
+        // update transient state in a side-effect to ensure that these changes only take effect
+        // after
+        // the command has been successfully processed
+        sideEffectWriter.appendSideEffect(
```

Comment:
I actually expected it to be more readable if I separate them per processor instead of updating couple of processors at once. 🤔  The correlate processor is appending `CORRELATED` which triggers `subscriptionState.updateToOpenedState()` and that method was already updated in the previous commit.